### PR TITLE
Use a default prefix for the mkstemp utils function

### PR DESCRIPTION
**Pull request by terminalmage**
This also moves this function into salt.utils.files so that alongside it can be an attribute (`TEMPFILE_PREFIX`) which can be used to get the default tempfile prefix. This allows for simpler logic for properly cleaning the tempfiles created by the `__clean_tmp()` function in `salt.modules.file`.

Refs: https://github.com/saltstack/salt/issues/37021

This should be merged before https://github.com/saltstack/salt/pull/37023. https://github.com/saltstack/salt/pull/37023 will cause merge conflicts with this PR.

This PR is a better long-term fix for #37021, but touches a lot of files, so I thought the best course of action would be to make the minimal change in the 2015.8 branch, and then make the better long-term change in develop.

### CODE REVIEW

File: salt/modules/dockerng.py [hunk 1]
```diff
@@ -1,4 +1,4 @@
-# -*- coding: utf-8 -*-
+      # -*- coding: utf-8 -*-
```
Comment (rallytime) [inline salt/modules/dockerng.py]: I don't think you intended this line change...

**Review by rallytime — DISMISSED** [review]
I had one tiny comment above, but it also looks like you have some small lint errors and some tests to look at. :(

Though I think that many of test failures will clear up once pylint is addressed.

**Comment by terminalmage** [comment]
OK, I pushed a lint change.

**Comment by rallytime** [comment]
Ok, just one failure now then. :)

The mock paths probably just need to be updated: https://jenkins.saltstack.com/job/PR/job/salt-pr-rs-cent7-n/6203/testReport/junit/unit.states.file_test/FileTestCase/test_managed/

**Comment by terminalmage** [comment]
@rallytime the remaining test failure is unrelated and fixed in https://github.com/saltstack/salt/pull/37033

**Pull request merged**